src/server/helpers/middleware.ts

```diff
@@ -32,25 +32,24 @@ interface $Request extends Request {
 	user: any
 }
 
-function makeLoader(modelName, propName, sortFunc?, relations = []) {
-	return function loaderFunc(req: $Request, res: $Response, next: NextFunction) {
-		const {orm}: any = req.app.locals;
-		const model = orm[modelName];
-		return model.fetchAll({withRelated: [...relations]})
-			.then((results) => {
-				const resultsSerial = results.toJSON();
-
-				res.locals[propName] =
-					sortFunc ? resultsSerial.sort(sortFunc) : resultsSerial;
-
-				next();
-
-				return null;
-			})
-			.catch(next);
-	};
+async function makeLoader(modelName, propName, sortFunc?, relations = []) {
+	try {
+		return async function loaderFunc(req: $Request, res: $Response, next: NextFunction) {
+			const {orm}: any = req.app.locals;
+			const model = orm[modelName];
+			const results = await model.fetchAll({withRelated: [...relations]});
+			const resultsSerial = results.toJSON();
+			res.locals[propName] =
+				sortFunc ? resultsSerial.sort(sortFunc) : resultsSerial;
+			next();
+			return null;
+		};
+	} catch (error) {
+		next(error);
+	}
```

Comment:
Oops, didn't notice that earlier, but the catch block has to be inside the returned middleware function of course.
Probably because I've reviewed the diff on GitHub, where the display isn't as comfortable as in a local editor.
```suggestion
	return async function loaderFunc(req: $Request, res: $Response, next: NextFunction) {
		try {
			const {orm}: any = req.app.locals;
			const model = orm[modelName];
			const results = await model.fetchAll({withRelated: [...relations]});
			const resultsSerial = results.toJSON();
			res.locals[propName] =
				sortFunc ? resultsSerial.sort(sortFunc) : resultsSerial;
			next();
			return null;
		} catch (error) {
			next(error);
		}
	};
```